code/modules/wiremod/shell/keyboard.dm

```diff
@@ -0,0 +1,54 @@
+/obj/item/keyboard_shell
+	name = "Keyboard Shell"
+	icon = 'icons/obj/wiremod.dmi'
+	icon_state = "setup_small_keyboard"
+	inhand_icon_state = "electronic"
+	worn_icon_state = "electronic"
+	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+	light_system = MOVABLE_LIGHT_DIRECTIONAL
+	light_on = FALSE
+
+/obj/item/keyboard_shell/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/shell, list(
+		new /obj/item/circuit_component/keyboard_shell()
+	), SHELL_CAPACITY_SMALL)
+
+/obj/item/circuit_component/keyboard_shell
+	display_name = "Keyboard Shell"
+	desc = "A handheld shell that allows the user to input a string. Use the shell in hand to open the input panel."
+
+	var/datum/port/output/signal
+	var/datum/port/output/entity
+	var/datum/port/output/output
+
+	var/illiterate
+
+/obj/item/circuit_component/keyboard_shell/populate_ports()
+	entity = add_output_port("User", PORT_TYPE_ATOM)
+	output = add_output_port("Message", PORT_TYPE_STRING)
+	signal = add_output_port("Signal", PORT_TYPE_SIGNAL)
+
+/obj/item/circuit_component/keyboard_shell/register_shell(atom/movable/shell)
+	RegisterSignal(shell, COMSIG_ITEM_ATTACK_SELF, .proc/send_trigger)
+
+/obj/item/circuit_component/keyboard_shell/unregister_shell(atom/movable/shell)
+	UnregisterSignal(shell, COMSIG_ITEM_ATTACK_SELF)
+
+/obj/item/circuit_component/keyboard_shell/proc/send_trigger(atom/source, mob/user)
+	SIGNAL_HANDLER
+	check_illiterate(user)
```

Comment:
just move this check into the use_keyboard proc, you don't need another proc for this